legacy/ui/legacy/src/main/java/com/fsck/k9/ui/settings/general/GeneralSettingsDataStore.kt

```diff
@@ -307,4 +309,9 @@ class GeneralSettingsDataStore(
         "move" -> SwipeAction.Move
         else -> throw AssertionError()
     }
+
+    private fun setTelemetryEnabled(enable: Boolean) {
+        K9.isTelemetryEnabled = enable
+        telemetryManager.setEnabled(enable)
```

Comment:
If the `TelemetryManager` could listen to changed settings, this would not be necessary.